### PR TITLE
Protectli vault jsl: UEFI payload

### PR DESCRIPTION
**Pull request by Stojek139808**

### CODE REVIEW

File: configs/config.protectli_pt201 [hunk 1]
```diff
@@ -1,4 +1,29 @@
 CONFIG_LOCALVERSION="4.12"
 CONFIG_VENDOR_PROTECTLI=y
+CONFIG_CBFS_SIZE=0x200000
+CONFIG_CONSOLE_CBMEM_BUFFER_SIZE=0x20000
+CONFIG_EDK2_BOOT_TIMEOUT=3
```
Comment (miczyg1) [inline configs/config.protectli_pt201:5]: Protectli uses 6 seconds

File: configs/config.protectli_pt201 [hunk 2]
```diff
@@ -1,4 +1,29 @@
 CONFIG_LOCALVERSION="4.12"
 CONFIG_VENDOR_PROTECTLI=y
+CONFIG_CBFS_SIZE=0x200000
```
Comment (miczyg1) [inline configs/config.protectli_pt201:3]: Only 2 MB CBFS?

File: configs/config.protectli_pt201 [hunk 3]
```diff
@@ -1,4 +1,29 @@
...
+CONFIG_EDK2_BOOTSPLASH_FILE="3rdparty/dasharo-blobs/dasharo/bootsplash.bmp"
+# CONFIG_ON_DEVICE_ROM_LOAD is not set
+CONFIG_SUBSYSTEM_VENDOR_ID=0x0000
+CONFIG_SUBSYSTEM_DEVICE_ID=0x0000
+CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
+# CONFIG_SMMSTORE is not set
 CONFIG_DEFAULT_CONSOLE_LOGLEVEL_0=y
+CONFIG_POST_IO_PORT=0x80
+CONFIG_PAYLOAD_EDK2=y
+CONFIG_EDK2_REPO_CUSTOM=y
```
Comment (miczyg1) [inline configs/config.protectli_pt201:16]: This option is not needed I think

File: configs/config.protectli_pt201 [hunk 4]
```diff
@@ -1,4 +1,29 @@
 CONFIG_LOCALVERSION="4.12"
 CONFIG_VENDOR_PROTECTLI=y
+CONFIG_CBFS_SIZE=0x200000
+CONFIG_CONSOLE_CBMEM_BUFFER_SIZE=0x20000
+CONFIG_EDK2_BOOT_TIMEOUT=3
 CONFIG_BOARD_PROTECTLI_PT201=y
+CONFIG_EDK2_BOOTSPLASH_FILE="3rdparty/dasharo-blobs/dasharo/bootsplash.bmp"
```
Comment (miczyg1) [inline configs/config.protectli_pt201:7]: It needs protectli logo

File: payloads/external/Makefile.inc [hunk 5]
```diff
@@ -165,26 +165,43 @@ $(obj)/UEFIPAYLOAD.fd: $(DOTCONFIG) $(IPXE_EFI)
 		CONFIG_EDK2_REPOSITORY=$(CONFIG_EDK2_REPOSITORY) \
 		CONFIG_EDK2_TAG_OR_REV=$(CONFIG_EDK2_TAG_OR_REV) \
 		CONFIG_EDK2_UEFIPAYLOAD=$(CONFIG_EDK2_UEFIPAYLOAD) \
-		CONFIG_EDK2_REPO_OFFICIAL=$(CONFIG_EDK2_REPO_OFFICIAL) \
-		CONFIG_EDK2_REPO_MRCHROMEBOX=$(CONFIG_EDK2_REPO_MRCHROMEBOX) \
-		CONFIG_EDK2_REPO_CUSTOM=$(CONFIG_EDK2_REPO_CUSTOM) \
```
Comment (miczyg1) [inline payloads/external/Makefile.inc:170]: Why is this removed?

File: payloads/external/Makefile.inc [hunk 6]
```diff
@@ -165,26 +165,43 @@ $(obj)/UEFIPAYLOAD.fd: $(DOTCONFIG) $(IPXE_EFI)
...
 		CONFIG_EDK2_CBMEM_LOGGING=$(CONFIG_EDK2_CBMEM_LOGGING) \
 		CONFIG_EDK2_FOLLOW_BGRT_SPEC=$(CONFIG_EDK2_FOLLOW_BGRT_SPEC) \
 		CONFIG_EDK2_FULL_SCREEN_SETUP=$(CONFIG_EDK2_FULL_SCREEN_SETUP) \
 		CONFIG_EDK2_HAVE_EFI_SHELL=$(CONFIG_EDK2_HAVE_EFI_SHELL) \
+		CONFIG_EDK2_IPXE=$(CONFIG_EDK2_ENABLE_IPXE) \
+		CONFIG_EDK2_IPXE_OPTION_NAME=$(CONFIG_EDK2_IPXE_OPTION_NAME) \
+		CONFIG_EDK2_OPAL_PASSWORD=$(CONFIG_EDK2_OPAL_PASSWORD) \
 		CONFIG_EDK2_PRIORITIZE_INTERNAL=$(CONFIG_EDK2_PRIORITIZE_INTERNAL) \
 		CONFIG_EDK2_PS2_SUPPORT=$(CONFIG_EDK2_PS2_SUPPORT) \
+		CONFIG_EDK2_SATA_PASSWORD=$(CONFIG_EDK2_SATA_PASSWORD) \
```
Comment (miczyg1) [inline payloads/external/Makefile.inc:190]: Setup password is missing

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: payloads/external/Makefile.inc (hunk 5 above)
Comment (Stojek139808) [reply payloads/external/Makefile.inc:170]: What happend is I've copied all these params from other protectli board since some were missing and went over the ones that were deleted by the "copy-paste". The thing is, these three options are in kconfig, but are not used in makefile, which is why I've pushed the change the way it is. For example same goes for our ehl vault branch.

File: configs/config.protectli_pt201 (hunk 3 above)
Comment (Stojek139808) [reply configs/config.protectli_pt201:16]: Yes, the comment below explains why, will remove that.

File: configs/config.protectli_pt201 (hunk 2 above)
Comment (Stojek139808) [reply configs/config.protectli_pt201:3]: I am pretty sure this was added by `savedefconfig`, what should the value be?

File: configs/config.protectli_pt201 (hunk 1 above)
Comment (Stojek139808) [reply configs/config.protectli_pt201]: Fixed.

File: configs/config.protectli_pt201 (hunk 4 above)
Comment (Stojek139808) [reply configs/config.protectli_pt201]: Changed it to protectli, but the problem is we don't have jsp on blobs repository yet, and the one in `vault/` fails to build (BMP compression error in edk2).

File: payloads/external/Makefile.inc (hunk 6 above)
Comment (Stojek139808) [reply payloads/external/Makefile.inc:190]: What setup password? I've looked at other boards and I only see `CONFIG_EDK2_OPAL_PASSWORD` and `CONFIG_EDK2_SATA_PASSWORD`.
